### PR TITLE
Update pool for release

### PR DESCRIPTION
**Pull request by mmitche**

### CODE REVIEW

File: eng/PrepareRelease.yml [hunk 1]
```diff
@@ -8,7 +8,8 @@ stages:
   - job: PrepareReleaseJob
     displayName: Prepare release with Darc
     pool: 
-      vmImage: windows-latest
+      name: NetCore1ESPool-Internal
+      demands: ImageOverride -equals build.windows.10.amd64.vs2017
```
Comment (hoyosjs) [inline eng/PrepareRelease.yml:12]: Is 2017 preferred to 2022 or 2019?

Comment (mmitche) [reply eng/PrepareRelease.yml:12]: Doesn't matter. Just has to run a little bit of a build. I think the other parts of this repo were using 2017.

Comment (jander-msft) [reply eng/PrepareRelease.yml:12]: All other build legs are using VS2019

**Review by hoyosjs — APPROVED** [review]

File: eng/PrepareRelease.yml [hunk 2]
```diff
@@ -8,7 +8,8 @@ stages:
   - job: PrepareReleaseJob
     displayName: Prepare release with Darc
     pool: 
-      vmImage: windows-latest
+      name: NetCore1ESPool-Internal
+      demands: ImageOverride -equals build.windows.10.amd64.vs2017
     variables:
     - ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest'), or(startsWith(variables['Build.SourceBranch'], 'refs/heads/release/'), startsWith(variables['Build.SourceBranch'], 'refs/heads/internal/release/'), startsWith(variables['Build.SourceBranch'], 'refs/heads/test/release/'))) }}:
```
[the remark this answers is not in the document]
Comment (jander-msft) [reply eng/PrepareRelease.yml:18]: I applied the same logic we use everywhere else where in the public project it uses the public pool and for non-public project it uses the internal pool.

**Pull request merged**